### PR TITLE
feat: Log API calls to logdna in production only

### PR DESCRIPTION
**Pull request by DavidLiang01**
To test this PR:

1. Log into [logdna](https://app.logdna.com/account/signin) with credentials (see discord for secrets)
2. Create .env file in server folder
3. Add environment variable in .env file with key=value as follows: LOGDNA= Logdna ingestion key (see discord secrets)
4. Make an API call 
5. Nothing should appear in Logdna

To log an api call to logdna, do the following:

1. Go to default.js file in server folder
2. Change line 12 `zeetEnv: process.env.ZEET_ENVIRONMENT || 'local',` to `zeetEnv: process.env.ZEET_ENVIRONMENT || 'develop',`

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-kuyqq8 Integrate a third party logging manager](https://app.clickup.com/t/kuyqq8)

File: server/app/App.ts [hunk 1]
```diff
@@ -33,6 +36,27 @@ export class App {
...
+          app: 'backend service',
+          env: this.config.get<string>('zeetEnv'),
+          handleExceptions: true,
+          level: 'info',
+        },
+      };
+
+      const loggerTransports = [new winston.transports.Console(options.console)];
+
+      if (config.get<string>('zeetEnv').toLowerCase().includes('prod')) {
```
Comment (Mackbellemore) [inline server/app/App.ts]: @gordonpn doesnt this match on the branch name? so it would be main

File: server/app/App.ts [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 import winston, { Logger } from 'winston';
 import expressWinston from 'express-winston';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore because types don't exist for this library
+import LogdnaWinston from 'logdna-winston';
 import { InversifyExpressServer } from 'inversify-express-utils';
 import { container } from './registry';
 import { Application } from 'express';
-import { IConfig } from 'config';
+import config, { IConfig } from 'config';
```
Comment (Mackbellemore) [inline server/app/App.ts]: I dont think this import of config is needed

File: server/app/registry.ts [hunk 3]
```diff
@@ -36,6 +39,18 @@ logger.add(
   })
 );
 
+if (config.get<string>('zeetEnv').toLowerCase().includes('prod')) {
```
Comment (gordonpn) [inline server/app/registry.ts]: zeetEnv will never contain `prod` in the case expected. zeetEnv is based on the branch that it is built off of, such as `develop` or `main`.

**Review by Mackbellemore — COMMENTED** [review]
the package-lock change in client is a bit weird

File: server/app/registry.ts (hunk 3 above)
Comment (DavidLiang01) [reply server/app/registry.ts]: so I put develop and main?

**Comment by gordonpn** [comment]
> the package-lock change in client is a bit weird

I might have installed and uninstalled something by mistake. I can't remember lol

File: server/app/registry.ts (hunk 3 above)
Comment (gordonpn) [reply server/app/registry.ts]: Yeah, we'll log to LogDNA if the zeetEnv matches develop or main

Comment (Mackbellemore) [reply server/app/registry.ts]: I think we should only log main, it will be confusing if not

Comment (Mackbellemore) [reply server/app/registry.ts]: unless we can have two different views for the logs

File: server/app/App.ts [hunk 4]
```diff
@@ -50,10 +74,7 @@ export class App {
         server.use(bodyParser.json());
         server.use(
           expressWinston.logger({
-            transports: [new winston.transports.Console()],
-            meta: false,
-            expressFormat: true,
-            statusLevels: true,
```
Comment (gordonpn) [inline server/app/App.ts]: Why did you remove these? I'm curious

I think it would be cool to see metadata from the Express logger logged to LogDNA when the backend is running in main

File: server/app/registry.ts (hunk 3 above)
Comment (DavidLiang01) [reply server/app/registry.ts]: We can log when we are in main only

Comment (gordonpn) [reply server/app/registry.ts]: Sure, main only works for me

Comment (gordonpn) [reply server/app/registry.ts]: > unless we can have two different views for the logs

Yeah that's possible as well by creating two "app" like "backend service develop" and "backend service main"

Comment (DavidLiang01) [reply server/app/registry.ts:42]: so what do you guys want?

Comment (DavidLiang01) [reply server/app/registry.ts]: I will put 'main' only

Comment (gordonpn) [reply server/app/registry.ts]: What if one day we want to debug something from our develop branch using the logs?

I like the idea of having two views just in case

Comment (DavidLiang01) [reply server/app/registry.ts]: but we are logging in the console

Comment (gordonpn) [reply server/app/registry.ts]: > but we are logging in the console

When the backend is hosted on Zeet? Only Mack has access to the runtime logs. So if we log to LogDNA, everyone could have access to the runtime logs

**Comment by gordonpn** [comment]
Take the LogDNA secret from the Discord secrets channel and add that into a `.env` file in the server directory as follows:

```bash
LOGDNA=f________d
```

Change line `zeetEnv: process.env.ZEET_ENVIRONMENT || 'local',` to `zeetEnv: process.env.ZEET_ENVIRONMENT || 'develop',`

On https://app.logdna.com/ you will be able to filter by "App", `undefined backend service` are logs that come from a local dev setup otherwise it will be `develop backend service` or `main backend service`

**Review by Mackbellemore — APPROVED** [review]
:racing_car:

**Comment by DavidLiang01** [comment]
Previously, you guys mentioned that, for zeet, the environment variable can be `develop` or `main`. Can it be something else? If it is either `develop `or `main`, then we don't need to check the condition to add the logdna transporter into the  loggerTransports since we are always logging. Right?

**Comment by gordonpn** [comment]
> Previously, you guys mentioned that, for zeet, the environment variable can be `develop` or `main`. Can it be something else? If it is either `develop `or `main`, then we don't need to check the condition to add the logdna transporter into the  loggerTransports since we are always logging. Right? 

The Zeet env is based off of the branch name

**Review by Ming424 — APPROVED** [review]
approved by thierry
